clang/lib/Driver/Driver.cpp

```diff
@@ -5524,9 +5524,11 @@ InputInfo Driver::BuildJobsForActionNoCache(
   else {
     std::string OffloadingPrefix;
     // When generating binaries with -fsycl-link-target or -fsycl-link, the
-    // output file prefix is the triple arch only.
+    // output file prefix is the triple arch only.  Do not add the arch when
+    // compiling for FPGA with -fsycl-link.
     if (Args.getLastArg(options::OPT_fsycl_link_targets_EQ) ||
-        Args.hasArg(options::OPT_fsycl_link_EQ)) {
+        (Args.hasArg(options::OPT_fsycl_link_EQ) &&
+         !Args.hasArg(options::OPT_fintelfpga))) {
```

Comment:
If this needs to be FPGA-specific, then
```suggestion
         TC->getTriple().getSubArch() != llvm::Triple::SPIRSubArch_fpga)) {
```
could be more general.

Reply:
The target arch is really only needed when we know we are outputting for a device.  In the case here for -fsycl-link and FPGA, the output is host (as seen by the file name).  We can make this more generic by keying off of that.  I make adjustments to reflect.

Reply:
The adjustments look good!